profile/src/main/java/com/scottlogic/deg/profile/reader/file/CsvInputStreamReader.java

```diff
@@ -24,26 +27,25 @@
 import java.io.InputStream;
 import java.io.UncheckedIOException;
 import java.nio.charset.Charset;
-import java.util.HashSet;
 import java.util.List;
-import java.util.Set;
+import java.util.stream.Collectors;
 
 public final class CsvInputStreamReader {
 
     private CsvInputStreamReader() {
         throw new UnsupportedOperationException("No instantiation of static class");
     }
 
-    public static Set<String> retrieveLines(InputStream stream) {
+    public static DistributedSet<String> retrieveLines(InputStream stream) {
         List<CSVRecord> records = parse(stream);
+        return new FrequencyDistributedSet<>(records.stream()
+            .map(CsvInputStreamReader::createWeightedElement)
+            .collect(Collectors.toSet()));
+    }
 
-        Set<String> firstElementFromEachRecord = new HashSet<>();
-        for (CSVRecord record : records) {
-            String firstElement = firstElementFromRecord(record);
-            firstElementFromEachRecord.add(firstElement);
-        }
-
-        return firstElementFromEachRecord;
+    private static WeightedElement<String> createWeightedElement(CSVRecord record) {
+        final double weight = record.size() > 1 ? Double.parseDouble(record.get(1)) : 1.0F;
```

Comment:
real minor opinion based question.

do you think this would look better as  
``` record.size() == 1 ? 1.0F : Double.parseDouble(record.get(1))```

Reply:
I don't - I prefer to have the expected or typical case on the left, and the exceptional case on the right. My logic being having the typical positive case reads better in `if(...)` statements than the negative case.

I'm keen to hear the other viewpoint here though!